### PR TITLE
Add fake v1alpha2 and keep v1alpha1 the only served version

### PR DESCRIPTION
**Pull request by yiqigao217**
Make a copy of the v1alpha1 api folder and change it into v1alpha2 with
the exact same schema. Make v1alpha1 the only served version and the
storage version. Make v1alpha2 unserved.

The `kubebuilder:unservedversion` marker seems not working, so this
commit used `kubebuilder:skipversion` on v1alpha2 CRDs. Without this
marker, v1alpha2 had `served: true` and `storage: false`. When I looked
at the CRs, I saw v1alpha2 version is served.

We don't want to serve v1alpha2 before it's ready, so we will only
manually change it to be servedversion for testing during development.

Part of #868 
The issue with `unservedversion` marker is filed [here](https://github.com/kubernetes-sigs/controller-tools/issues/462).

### CODE REVIEW

File: incubator/hnc/api/v1alpha2/hierarchy_types.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+)
+
+// ClearConditionCriteria is initialized in init(). See ClearConditionCriterion for more details.
+var ClearConditionCriteria map[Code]ClearConditionCriterion
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// +kubebuilder:object:root=true
+// +kubebuilder:skipversion
```
Comment (yiqigao217) [inline incubator/hnc/api/v1alpha2/hierarchy_types.go:95]: This is the `skipversion` marker added to the v1alpha2 copy from v1alpha1. The `unservedversion` seemed not working, see [issue](https://github.com/kubernetes-sigs/controller-tools/issues/462).

**Comment by yiqigao217** [comment]
> Why do you want to start with a full copy of _all_ the files, even the ones that will change over time? Did you consider only adding an empty skeleton for, say, HC but not adding anything else yet?

Because the changes are minor based on the v1alpha1 and I felt it's easier to review for later PR to make a diff on top of that, so reviewer can just see the changes instead of to review the newly-added code (which is changes upon v1alpha1).

**Comment by adrianludwin** [comment]
Makes sense

/lgtm
/approve

**Comment by yiqigao217** [comment]
/retest

The previous failed test logs are put in a gist and #871 is filed to investigate this failed test since it doesn't seem relevant.

**Pull request merged**